torch/autograd/function.py

```diff
@@ -533,12 +534,29 @@ def vmap(info, in_dims, *args):
 
     @classmethod
     def apply(cls, *args, **kwargs):
+        def bind_default_args(func, is_def_ctx, *args, **kwargs):
+            arguments: List[Any]
+
+            arguments = list(args)
+            # Used to bind to ctx
+            if is_def_ctx:
```

Comment:
We don't have to go through this binding logic if setup context isn't defined. That would remove all of the `is_def_ctx` logic here.

Reply:
Thank you for your advice, updated it.